server/job.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"github.com/google/uuid"
+	pb "github.com/romanlevin/felek/jobs"
+	"log"
+	"os/exec"
+	"sync"
+)
+
+type job struct {
+	id        string    // A UUID string
+	cmd       *exec.Cmd // The underlying Cmd object wrapping the job process
+	stopped   bool      // Has this job been stopped by a client?
+	exitError error     // The error returned by Wait on the Cmd
+	owner     string    // The username of the user who started the job
+	lock      *sync.RWMutex
+}
+
+// exitedWithExitCode assumes a lock is held
+func (j *job) exitedWithExitCode() bool {
```

Comment:
Nit: I'd name these `exitedWithExitCodeLocked()` and `runningLocked()` to indicate they're supposed to be invoked under lock.